### PR TITLE
add allowedUrls list for sentry

### PR DESCRIPTION
**Pull request by bethshook**
for: openstax/unified#1553

### CODE REVIEW

File: src/helpers/Sentry.ts [hunk 1]
```diff
@@ -38,6 +38,14 @@ export default {
...
         release: normalize(`rex@${config.RELEASE_ID}`),
         tracesSampleRate: 0.1,
+        whitelistUrls: [
+          /https?:\/\/localhost:3000/,
+          /https?:\/\/(www\.)?openstax\.org/,
+          /https?:\/\/dev\.openstax\.org/,
+          /https?:\/\/(.*\.)?sandbox\.openstax\.org/,
+          /https?:\/\/staging\.openstax\.org/,
+          /https?:\/\/rex-web\.herokuapp\.com/,
+        ],
```
Comment (bethshook) [inline src/helpers/Sentry.ts:48]: @TomWoodward I'm probably missing things...

File: src/helpers/Sentry.ts [hunk 2]
```diff
@@ -38,6 +38,14 @@ export default {
         ],
         release: normalize(`rex@${config.RELEASE_ID}`),
         tracesSampleRate: 0.1,
+        whitelistUrls: [
+          /https?:\/\/localhost:3000/,
+          /https?:\/\/(www\.)?openstax\.org/,
+          /https?:\/\/dev\.openstax\.org/,
+          /https?:\/\/(.*\.)?sandbox\.openstax\.org/,
+          /https?:\/\/staging\.openstax\.org/,
```
Comment (TomWoodward) [inline src/helpers/Sentry.ts:46]: i'd combine these into _anything_ dot openstax dot org

File: src/helpers/Sentry.ts [hunk 3]
```diff
@@ -38,6 +38,14 @@ export default {
...
+          /https?:\/\/rex-web\.herokuapp\.com/,
```
Comment (TomWoodward) [inline src/helpers/Sentry.ts:47]: missing the PR environments on heroku

**Review by TomWoodward — CHANGES_REQUESTED** [review]

File: src/helpers/Sentry.ts [hunk 4]
```diff
@@ -27,6 +27,11 @@ export default {
   initializeWithMiddleware(): Middleware {
     return (store) => {
       Sentry.init({
+        allowUrls: [
+          /localhost/,
+          /openstax.org/,
+          /https?:\/\/rex-web(.*\.)?\.herokuapp\.com/,
+        ],
```
Comment (bethshook) [inline src/helpers/Sentry.ts:34]: @TomWoodward I'm still confused about what this list is doing exactly. Some errors are definitely still reported for URLs not on the list (ie when I remove `localhost`), but it seems like errors that are manually reported via `Sentry.captureException` are filtered out.

Would it be worth first trying some custom event filters [here](https://sentry.io/settings/openstax/projects/rex/fi) before we rely on a whitelist or does that seem too limiting/unpredictable?

File: src/helpers/Sentry.ts [hunk 5]
```diff
@@ -27,6 +27,11 @@ export default {
   initializeWithMiddleware(): Middleware {
     return (store) => {
       Sentry.init({
+        allowUrls: [
+          /localhost/,
+          /openstax.org/,
+          /https?:\/\/rex-web(.*\.)?\.herokuapp\.com/,
```
Comment (PiotrKozlowski) [inline src/helpers/Sentry.ts:33]:
```suggestion
          /https?:\/\/rex-web(.*)?\.herokuapp\.com/,
```

File: src/helpers/Sentry.ts (hunk 4 above)
Comment (PiotrKozlowski) [reply src/helpers/Sentry.ts:34]: [It seems like `allowURL` works only for captured events](whitelistUrls)
I think that maybe we want to use [`whitelistUrls`](https://docs.sentry.io/clients/javascript/tips/#decluttering-sentry)?

**Review by PiotrKozlowski — CHANGES_REQUESTED** [review]

File: src/helpers/Sentry.ts (hunk 4 above)
Comment (jarosik10) [reply src/helpers/Sentry.ts:34]: [docs](https://docs.sentry.io/platforms/javascript/guides/angular/configuration/filtering/#decluttering-sentry):
_Prior to version 5.17.0, allowUrls and denyUrls were called whitelistUrls and blacklistUrls respectively. These options are still supported for backwards compatibility reasons, but they will be removed in version 6.0._ 

What is the reason behind switching to `allowUrls`?

Comment (jarosik10) [reply src/helpers/Sentry.ts:34]: Okay, i see:)

**Review by jarosik10 — APPROVED** [review]

**Review by TomWoodward — APPROVED** [review]

**Review by PiotrKozlowski — APPROVED** [review]

**Pull request merged**